Add documentation for delete protection for instances & volumes (#433)
apache/cloudstack-documentation: source/adminguide/virtual_machines.rst

      <i.n> represents the value of the global configuration - instance.name

Instance delete protection
--------------------------

> CloudStack provides a feature to protect Instances from accidental deletion.

Reviewer: ```suggestion
CloudStack protects instances from accidental deletion using a delete protection flag, which is false by default.
```

When delete protection is enabled for an instance, it cannot be deleted through

> the UI or API. It can only be delete after removing delete protection from the

Reviewer: ```suggestion
the UI or API. It can only be deleted after removing delete protection from the
```

      <i.n> represents the value of the global configuration - instance.name

Instance delete protection
--------------------------

CloudStack protects instances from accidental deletion using a delete protection
flag, which is false by default. When delete protection is enabled for an
instance, it cannot be deleted through the UI or API. It can only be deleted
after removing delete protection from the instance.

Delete protection can be enabled for an instance via updateVirtualMachine API.

.. code:: bash

>    cmk update virtualmachine id=<instance id> deletionprotection=true

Reviewer: ```suggestion
   cmk update virtualmachine id=<instance id> deleteprotection=true
```
